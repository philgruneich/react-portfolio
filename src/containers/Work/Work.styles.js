import styled from 'styled-components';
import { Title } from '~/components/Title';
import { fluidType, pxToEm } from '~/utils/cssMixins';

export const WorkArticle = styled.article`
  display: grid;
  grid-gap: var(--xMargin)  calc(var(--xMargin) * 2);
  padding: var(--yMargin) var(--xMargin);
  background: #ff8361;
  color: #fff;

  @media (min-width: 992px) {
    grid-template-columns: [list] 1fr [text] 1fr;
  }

  ${props => props.hasHighContrast && `
    background: #000;
  `}
`;

export const WorkHeader = styled.header`
  @media (min-width: 992px) {
    grid-column-end: span 2;
  }
`;

export const WorkTitle = styled(Title)`
  color: #fdd630;
  font-weight: 700;

  ${props => props.hasHighContrast && `
    color: #fff;
  `}
`;

export const WorkAside = styled.aside`
  line-height: 1.38889;
  font-size: ${fluidType({
    viewportWidth: {
      min: 320,
      max: 1200
    },
    size: {
      min: 20,
      max: 36
    }
  })};

  @media(min-width: 1200px) {
    font-size: ${pxToEm(36)};
  }
`;

export const WorkText = styled.p`
  margin: 0;
`;

export const WorkList = styled.ul`
  margin: 0;
  padding: 0;
  list-style: none;
`;

export const WorkListItem = styled.li`
  font-weight: 700;
  line-height: 1.46841;
  font-size: ${fluidType({
    viewportWidth: {
      min: 320,
      max: 1200
    },
    size: {
      min: 32,
      max: 64
    }
  })};

  @media(min-width: 1200px) {
    font-size: ${pxToEm(64)};
    line-height: 1.25;
  }
`;

export const WorkSection = styled.section``;
