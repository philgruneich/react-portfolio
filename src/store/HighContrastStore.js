import React, { useState, createContext } from 'react';

export const HighContrastContext = createContext();
HighContrastContext.displayName = 'HighContrastContext';

export const HighContrastStore = props => {
  const [hasContrast, saveHasContrast] = useState(false);

  const toggleContrast = () => {
    saveHasContrast(() => !hasContrast);
  }

  return (
    <HighContrastContext.Provider
      value={{
        state: hasContrast,
        dispatch: saveHasContrast,
        toggle: toggleContrast
      }}
      {...props}
    />
  )
}
