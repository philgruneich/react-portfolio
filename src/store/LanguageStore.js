import React, { useState, useEffect, createContext } from 'react';
import { languages } from '~/lang';

export const LanguageContext = createContext();
LanguageContext.displayName = 'LanguageContext';

export const LanguageStore = props => {
  const [lang, saveLang] = useState('en');
  const [translations, saveTranslations] = useState(languages[lang]);

  useEffect(() => {
    saveTranslations(languages[lang]);
  }, [lang]);

  return (
    <LanguageContext.Provider
      value={{
        state: { lang, translations },
        dispatch: saveLang
      }}
      {...props}
    />
  )
}
