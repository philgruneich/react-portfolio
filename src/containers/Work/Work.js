import React, { useContext } from 'react';
import { LanguageContext } from '~/store/LanguageStore';
import { HighContrastContext } from '~/store/HighContrastStore';
import {
  WorkArticle,
  WorkHeader,
  WorkTitle,
  WorkAside,
  WorkText,
  WorkList,
  WorkListItem,
  WorkSection
} from './Work.styles';

export const Work = () => {
  const langContext = useContext(LanguageContext);
  const contrastContext = useContext(HighContrastContext);

  const _ = langContext.state.translations.work;

  const clients = Clients.map(c => <WorkListItem key={c}>{c}</WorkListItem>);

  return (
    <WorkArticle hasHighContrast={contrastContext.state}>
      <WorkHeader>
        <WorkTitle hasHighContrast={contrastContext.state}>{_.title}</WorkTitle>
      </WorkHeader>
      <WorkAside>
        <WorkText>{_.text}</WorkText>
      </WorkAside>
      <WorkSection>
        <WorkList>
          {clients}
        </WorkList>
      </WorkSection>
    </WorkArticle>
  )
}

const Clients = [
  'O Boticário',
  'Renault',
  'Frimesa',
  'Apae',
  'Corteva',
  'Portinari',
  'Uniprime',
  'HouseCricket',
  'Shopping Mueller',
  'TFG',
  'Bem Paraná'
];
