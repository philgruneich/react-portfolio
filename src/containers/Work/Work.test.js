import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { LanguageStore } from '~/store/LanguageStore';
import { HighContrastStore } from '~/store/HighContrastStore';
import { Work } from './Work';

const defaultLanguageState = {
  state: {
    lang: 'en',
    translations: {
      work: {
        title: 'Title',
        text: 'Text'
      }
    }
  },
  dispatch: jest.fn()
};

const defaultHighContrastState = {
  state: false
};

test('Matches snapshot', () => {
  const { asFragment } = render(
    <LanguageStore value={defaultLanguageState}>
      <HighContrastStore value={defaultHighContrastState}>
        <Work />
      </HighContrastStore>
    </LanguageStore>
  );

  expect(asFragment()).toMatchSnapshot();
});

test('Matches snapshot with high contrast', () => {
  const { asFragment } = render(
    <LanguageStore value={defaultLanguageState}>
      <HighContrastStore value={{
        state: true
      }}>
        <Work />
      </HighContrastStore>
    </LanguageStore>
  );

  expect(asFragment()).toMatchSnapshot();
});
