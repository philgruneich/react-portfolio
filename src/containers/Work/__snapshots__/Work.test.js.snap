// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<DocumentFragment>
  <article
    class="Workstyles__WorkArticle-v7yxvq-0 cbLtt"
  >
    <header
      class="Workstyles__WorkHeader-v7yxvq-1 hYwSwS"
    >
      <h2
        class="Title-sc-1izy8tn-0 Workstyles__WorkTitle-v7yxvq-2 dvoTJc"
      >
        Title
      </h2>
    </header>
    <aside
      class="Workstyles__WorkAside-v7yxvq-3 hVeXEE"
    >
      <p
        class="Workstyles__WorkText-v7yxvq-4 bwqqOW"
      >
        Text
      </p>
    </aside>
    <section
      class="Workstyles__WorkSection-v7yxvq-7 eEPcRS"
    >
      <ul
        class="Workstyles__WorkList-v7yxvq-5 fLDfAf"
      >
        <li
          class="Workstyles__WorkListItem-v7yxvq-6 gMwSix"
        >
          O Botic치rio
        </li>
        <li
          class="Workstyles__WorkListItem-v7yxvq-6 gMwSix"
        >
          Renault
        </li>
        <li
          class="Workstyles__WorkListItem-v7yxvq-6 gMwSix"
        >
          Frimesa
        </li>
        <li
          class="Workstyles__WorkListItem-v7yxvq-6 gMwSix"
        >
          Apae
        </li>
        <li
          class="Workstyles__WorkListItem-v7yxvq-6 gMwSix"
        >
          Corteva
        </li>
        <li
          class="Workstyles__WorkListItem-v7yxvq-6 gMwSix"
        >
          Portinari
        </li>
        <li
          class="Workstyles__WorkListItem-v7yxvq-6 gMwSix"
        >
          Uniprime
        </li>
        <li
          class="Workstyles__WorkListItem-v7yxvq-6 gMwSix"
        >
          HouseCricket
        </li>
        <li
          class="Workstyles__WorkListItem-v7yxvq-6 gMwSix"
        >
          Shopping Mueller
        </li>
        <li
          class="Workstyles__WorkListItem-v7yxvq-6 gMwSix"
        >
          TFG
        </li>
        <li
          class="Workstyles__WorkListItem-v7yxvq-6 gMwSix"
        >
          Bem Paran치
        </li>
      </ul>
    </section>
  </article>
</DocumentFragment>
`;

exports[`Matches snapshot with high contrast 1`] = `
<DocumentFragment>
  <article
    class="Workstyles__WorkArticle-v7yxvq-0 gzKHpN"
  >
    <header
      class="Workstyles__WorkHeader-v7yxvq-1 hYwSwS"
    >
      <h2
        class="Title-sc-1izy8tn-0 Workstyles__WorkTitle-v7yxvq-2 ebWSGx"
      >
        Title
      </h2>
    </header>
    <aside
      class="Workstyles__WorkAside-v7yxvq-3 hVeXEE"
    >
      <p
        class="Workstyles__WorkText-v7yxvq-4 bwqqOW"
      >
        Text
      </p>
    </aside>
    <section
      class="Workstyles__WorkSection-v7yxvq-7 eEPcRS"
    >
      <ul
        class="Workstyles__WorkList-v7yxvq-5 fLDfAf"
      >
        <li
          class="Workstyles__WorkListItem-v7yxvq-6 gMwSix"
        >
          O Botic치rio
        </li>
        <li
          class="Workstyles__WorkListItem-v7yxvq-6 gMwSix"
        >
          Renault
        </li>
        <li
          class="Workstyles__WorkListItem-v7yxvq-6 gMwSix"
        >
          Frimesa
        </li>
        <li
          class="Workstyles__WorkListItem-v7yxvq-6 gMwSix"
        >
          Apae
        </li>
        <li
          class="Workstyles__WorkListItem-v7yxvq-6 gMwSix"
        >
          Corteva
        </li>
        <li
          class="Workstyles__WorkListItem-v7yxvq-6 gMwSix"
        >
          Portinari
        </li>
        <li
          class="Workstyles__WorkListItem-v7yxvq-6 gMwSix"
        >
          Uniprime
        </li>
        <li
          class="Workstyles__WorkListItem-v7yxvq-6 gMwSix"
        >
          HouseCricket
        </li>
        <li
          class="Workstyles__WorkListItem-v7yxvq-6 gMwSix"
        >
          Shopping Mueller
        </li>
        <li
          class="Workstyles__WorkListItem-v7yxvq-6 gMwSix"
        >
          TFG
        </li>
        <li
          class="Workstyles__WorkListItem-v7yxvq-6 gMwSix"
        >
          Bem Paran치
        </li>
      </ul>
    </section>
  </article>
</DocumentFragment>
`;
